articles/libraries/lock-android/v2/custom-authentication-providers.md

```diff
@@ -3,17 +3,23 @@ section: libraries
 toc_title: Custom Authentication Providers
 description: Implementing custom authentication providers
 ---
-
 # Custom Authentication Providers
 
 **Auth0.Android** includes the `WebAuthProvider` class to handle the authorize flow using the Browser. But what if you want to use your own implementation or a Native version of an `AuthProvider`?
 
 ## The AuthProvider class
+
 Create a class that implements the `AuthProvider` interface and override its methods.
 
-You can also use any _Native_ implementation already provided by Auth0. Currently available are [Google](/libraries/lock-android/native-provider-google) and [Facebook](/libraries/lock-android/native-provider-facebook).
+You can also use any _Native_ implementation already provided by Auth0. The [native social authentication](/libraries/lock-android/v2/native-social-authentication) providers currently available are Google and Facebook.
+
+::: warning
+The native social providers rely on a deprecated grant type. Clients created after June 8th 2017 won't be able to use this feature.
+We recommend using browser-based flows, as explained in [Authentication with Auth0 Hosted Login Page](/libraries/auth0-android#authentication-with-auth0-hosted-login-page).
```

Comment:
Avoid using whitespace inside notes/warnings/panel, the end result doesn't look good.

In this case, it's ignored either way
![image](https://user-images.githubusercontent.com/11715799/30694490-e0dbe0c6-9edc-11e7-9d71-53864a7e4c7b.png)